components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggingHandler.java

```diff
@@ -111,22 +114,25 @@ private boolean isAuditLoggerUserNameEnabled(Event event) throws IdentityEventEx
 
     protected void doPublishAuthenticationStepSuccess(AuthenticationAuditData authenticationData) {
 
+        String username = getUsernameForAuditLog(authenticationData.getAuthenticatedUser());
         String auditData = "\"" + "ContextIdentifier" + "\" : \"" + authenticationData.getContextIdentifier()
-                + "\",\"" + "AuthenticatedUser" + "\" : \"" + authenticationData.getAuthenticatedUser()
+                + "\",\"" + "AuthenticatedUser" + "\" : \"" + username
                 + "\",\"" + "AuthenticatedUserTenantDomain" + "\" : \"" + authenticationData.getTenantDomain()
                 + "\",\"" + "ServiceProviderName" + "\" : \"" + authenticationData.getServiceProvider()
                 + "\",\"" + "RequestType" + "\" : \"" + authenticationData.getInboundProtocol()
                 + "\",\"" + "RelyingParty" + "\" : \"" + authenticationData.getRelyingParty()
                 + "\",\"" + "AuthenticatedIdP" + "\" : \"" + authenticationData.getAuthenticatedIdps()
                 + "\"";
         auditData = addContextualInfo(auditData, authenticationData);
-
+        /*
+        Here username is used for the initiator instead of userId when the log masking is enabled also, as userId is
+        not resolved yet.User is not authenticated yet.
```

Comment:
Here the userId cannot be resolved as the user is not authenticated yet.  Hence the username is used for the initiator even when the log masking is enabled.